packages/edit-site/src/components/global-styles/custom-css.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * WordPress dependencies
+ */
+import { TextareaControl, Panel, PanelBody } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useStyle } from './hooks';
+
+function CustomCSSControl() {
+	const [ customCSS, setCustomCSS ] = useStyle( 'css' );
+	const [ themeCSS ] = useStyle( 'css', null, 'base' );
+	const ignoreThemeCustomCSS = '/* IgnoreThemeCustomCSS */';
```

Comment:
Hi there! I was trying to refactor this control to be like all the other "style" UI components and stumbled upon this specific behavior for custom CSS, where it seems that we need to be aware of "base" styles and have some custom behavior tied to it. Can you explain it a bit further and also why it has to be this way, why don't we treat the "css" theme.json property like any other property?

Reply:
Some initial testing showed that it was confusing not being able to see what custom CSS may be set in the theme.json, eg. some custom CSS could be set by user and not apply as overridden by higher specificity selectors in theme.json.

Two approaches to exposing the theme.json custom CSS to the user:

- Display any theme.json CSS above the input box. A disadvantage of this is that in order to override any existing selectors the user needs to add selectors with a higher specificity
- Include the theme.json within the custom CSS input box so the user can remove/override simply by deleting it 

The second approach was decided on.

Reply:
> Some initial testing showed that it was confusing that any theme.json custom CSS was overridden completely by any custom CSS added as initially it took the same approach as other global styles in that user styles override the theme.json settings. For example, the theme.json CSS might set a heading font color, and the user may add a paragraph font family which will wipe the heading font color.

For me this is only confusing, if the CSS is not shown initially in the box.

Reply:
> So a decision was made to include the theme.json CSS in the global styles input box to allow the user to make a conscious choice about which ones to keep.

Re-reading this, it seems like you're saying that we're already showing the theme.json CSS in the box, which is what I would expect as well, but I guess the different maybe when we edit that CSS.

For me, after editing, we should only show the edited CSS (which may or may not contain some parts of the original theme.json CSS).